### PR TITLE
Use Docker multi-stage build.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Comment by Marenz** [comment]
for referencing/bookkeeping: 
fixes #7128

**Comment by chriseth** [comment]
Sorry for not referencing. I won't have time to work on this for a while, so please just take it over!

**Comment by Marenz** [comment]
This results in an image-size of 1.65 GB (original 1.82GB) and 11 layers (original 16 layers)

File: .circleci/docker/Dockerfile.ubuntu1904 [hunk 1]
```diff
@@ -21,7 +21,7 @@
 #
 # (c) 2016-2019 solidity contributors.
 #------------------------------------------------------------------------------
-FROM buildpack-deps:disco
+FROM buildpack-deps:disco as base
```
Comment (christianparpart) [inline .circleci/docker/Dockerfile.ubuntu1904]:
```suggestion
FROM buildpack-deps:disco AS base
```
as a matter of homogeneity, I'd propose to have all keywords capitalized. (including the "as" a few lines below)

File: .circleci/docker/Dockerfile.ubuntu1904 [hunk 2]
```diff
@@ -41,14 +41,7 @@ RUN set -ex; \
...
 
-# Aleth for end-to-end tests
-ARG ALETH_VERSION="1.6.0"
-ARG ALETH_HASH="7f7004e1563299bc57882e32b32e4a195747dfb6"
-ARG ALETH_URL="https://github.com/ethereum/aleth/releases/download/v${ALETH_VERSION}/aleth-${ALETH_VERSION}-linux-x86_64.tar.gz"
-RUN set -ex; \
-	wget -q -O /tmp/aleth.tar.gz "${ALETH_URL}"; \
-	test "$(shasum /tmp/aleth.tar.gz)" = "$ALETH_HASH  /tmp/aleth.tar.gz"; \
-	tar -xf /tmp/aleth.tar.gz -C /usr
+FROM base as libraries
```
Comment (christianparpart) [inline .circleci/docker/Dockerfile.ubuntu1904]:
```suggestion
FROM base AS libraries
```

File: .circleci/docker/Dockerfile.ubuntu1904 [hunk 3]
```diff
@@ -41,14 +41,7 @@ RUN set -ex; \
...
 	rm -rf /var/lib/apt/lists/*
 
-# Aleth for end-to-end tests
-ARG ALETH_VERSION="1.6.0"
-ARG ALETH_HASH="7f7004e1563299bc57882e32b32e4a195747dfb6"
-ARG ALETH_URL="https://github.com/ethereum/aleth/releases/download/v${ALETH_VERSION}/aleth-${ALETH_VERSION}-linux-x86_64.tar.gz"
-RUN set -ex; \
-	wget -q -O /tmp/aleth.tar.gz "${ALETH_URL}"; \
-	test "$(shasum /tmp/aleth.tar.gz)" = "$ALETH_HASH  /tmp/aleth.tar.gz"; \
-	tar -xf /tmp/aleth.tar.gz -C /usr
```
Comment (christianparpart) [inline .circleci/docker/Dockerfile.ubuntu1904]: This PR is about optimizing the docker image by utilizing stages. Why are we removing the Aleth binary here? (is EVMONE already enabled by default?) @chriseth :)

**Review by christianparpart — CHANGES_REQUESTED** [review]

File: .circleci/docker/Dockerfile.ubuntu1904 (hunk 3 above)
Comment (Marenz) [reply .circleci/docker/Dockerfile.ubuntu1904]: I added aleth back for now.

**Comment by Marenz** [comment]
Updated. A preview image is available at https://hub.docker.com/r/marenz/ethereum/tags for anyone to test.
Will publish it on ethereum docker hub when this PR is merged.

**Comment by Marenz** [comment]
Pushed to https://hub.docker.com/r/ethereum/solidity-buildpack-deps/tags tag `ubuntu1904`

**Pull request merged**